src/component/axis/AxisBuilder.ts

```diff
@@ -365,29 +366,47 @@ const builders: Record<'axisLine' | 'axisTickLabel' | 'axisName', AxisElementsBu
     },
 
     axisName(opt, axisModel, group, transformGroup) {
+        function calcDistanceToAxis() {
+            const defaultMargin = 10;
```

Comment:
We should probably use [xAxis.nameGap](https://echarts.apache.org/en/option.html#xAxis.nameGap) instead. The document can be updated so that when the `nameLocation` is `outsideXXX`, it means the distance between the axis name and the axis labels.